hnn_core/tests/test_drives.py

```diff
@@ -171,6 +171,51 @@ def test_add_drives():
         assert drive_conn['nc_dict']['A_weight'] == weights_ampa[target_type]
         assert drive_conn['nc_dict']['A_delay'] == syn_delays[target_type]
 
+    # Test probabalistic drive connections.
+    # drive with cell_specific=False
+    n_drive_cells = 10
+    probability = 0.5  # test that only half of possible connections are made
+    weights_nmda = {'L2_basket': 1.0, 'L2_pyramidal': 3.0, 'L5_pyramidal': 4.0}
```

Comment:
Following up from our previous conversation on this, it seems like this reduced connection may not be very practical since receptors would be defined with different weights as well. 

Also @rythorpe, all drive connections created from a single call share the same seed. So when the connection gets passed to `net.add_connection` with probability <1.0, the same connections will be preserved across receptors.

Reply:
sounds good. I'm a little uneasy with ensuring "sameness" with seeds. I worry that we'll get weird bugs. 

Alternative is to allow `add_connection` to accept list of receptors and then have [`connection_probability`](https://github.com/jonescompneurolab/hnn-core/blob/20b6903b0682473f2af20357e086e1a980f26712/hnn_core/network.py#L97) return the `conn_pairs` to drop, then drop those `conn_pairs` for both receptors. I realize it's a bit of work, so let's move on here. But worth keeping in mind if such an issue comes up in the future

Reply:
Very valid, moving to a separate issue

Reply:
It looks like `conn_seed` enforces the same "random" connection pattern across cell types as well. Thus, "random" connectivity is not statistically independent of other cell types (e.g., specifying 60% connection probability between a proximal evoked drive and L2/3 pyramidal cells will give the exact same connectivity pattern as for a 60% connection probability to L5 pyramidal cells). This seems sketchy to me. As a user generally speaking, I would by default assume that any pseudo-randomized process is free of structural correlations unless otherwise stated.

Reply:
This is a really good point, I'm putting this back to WIP until I add a cleaner way to handle this case.